### ISSUE TITLE
Writing binary is no longer executable

### ISSUE THREAD

**Issue by daephx** [RaafatTurki/hex.nvim#6]
WSL2 Ubuntu 22.04.2 LTS
NVIM v0.10.0-dev

I followed the exact steps as the gif in the README and attempting to write the buffer makes the binary no longer executable. 
Resulting an error of either `exec format error: ./a.out` or `Segmentation fault`.
This occurs without any modification to the data, simply writing the buffer after opening `a.out` had the same result.

**Comment by RaafatTurki** [comment]
Writing the buffer triggers a binary "rebuild" from the hex dump using xxd.

- Are you sure your `xxd` is properly working? (try using it to dump and revert a binary file)?
- Does this occur with other file types such as a png file?

**Comment by daephx** [comment]
I did some tests to find if there was any discrepancies.
The issue persists on WSL and on bare metal pop_OS, both with minimal config.
However, so far I can only reproduce the issue when my CWD is `~/.config/nvim`.
If I do the same exact steps somewhere else like a `/tmp` folder, the issue doesn't occur.

When writing the dump buffer the "Hello, World!" string get shifted ~20 lines downward, resulting in the exec format error.

**Comment by daephx** [comment]
> Are you sure your xxd is properly working? (try using it to dump and revert a binary file)?

I can't be sure, I'm not too familiar with the program.

> Does this occur with other file types such as a png file?

I did test a image file located at `~/Downloads/image.jpg` without issue.
I'm not sure if you or someone else could reproduce but it seems like some obscure behaviour that the rebuilding would not work in neovim config directory.

```bash
cd ~/.config/nvim
nvim -u minimal_config.lua
# Write hello.c
$ gcc hello.c
$ ./a.out
Hello, World!

# Modify "Hello, World!" and write hex
$ ./a.out
exec format error: ./a.out
```

**Comment by AlexZoe** [comment]
Hey, Raafat, I ran into the same issue and investigate a bit to see what's wrong.

It seems the conversion with `HexDump` inserts random bytes into the dump file, which obviously breaks the binary file when writing the buffer back to disk with `HexAssemble`.
This happens for both manual conversion and automatic one (i.e. `ReadBuf` and `WriteBuf`).

Using the same `xxd` commands manually in a terminal works without problems.
Converting a binary file to a hexdump in nvim also works directly in ExecMode (`%! xxd -g 1 -u <file>`).

NVim version used with LinuxMint 20.1.
```
NVIM v0.10.0-dev-289+gfa1baa9a4
Build type: Release
LuaJIT 2.1.0-beta3
```

I attached two screenshots below to show the output side by side (left: regular dump, right: HexDump).
The first image shows the differing sizes (hex offset at the bottom).
The second highlights the first line where random bytes got inserted.

I hope this helps.

![differing_size](https://github.com/RaafatTurki/hex.nvim/assets/50106176/629d1bff-141f-42b2-a4a9-3e22cc12006f)
![random_chars](https://github.com/RaafatTurki/hex.nvim/assets/50106176/e73b36d1-687e-4bb3-a131-d187843911a4)
